### PR TITLE
feat: adds capsized typography

### PR DESCRIPTION
**Pull request by madou**
- Brings `capsize` calculations into the `Heading` component.
- Creates a `Text` component for all other usages
- Cleans up typography usage and removes outliers

**TODO**

- [x] Capsize remainder sizes in `Text` component

### CODE REVIEW

File: packages/docs/src/components/side-nav.tsx [hunk 1]
```diff
@@ -41,9 +40,7 @@ const StyledLink = styled(Link)`
   }
 
   &[aria-current='page'] {
-    border-left: 2px solid ${colors.primary};
-    padding-left: 10px;
-    margin-left: -12px;
+    color: ${colors.primary};
```
Comment (lorvsso) [inline packages/docs/src/components/side-nav.tsx:43]: nit: this is a minor accessibility regression, using color alone to distinguish meaning isn't as usable as the "indicator" you had before. colors.primary has sufficient contrast against the background color but not enough to the standard text color

File: packages/docs/src/pages/what-is-compiled.mdx [hunk 2]
```diff
@@ -8,7 +8,7 @@ import { BeforeHowItWorks, AfterHowItWorks } from '../examples/how-it-works';
 
 # What is Compiled?
 
-Compiled is a _build time_ [atomic CSS](/atomic-css)-in-JS library that has a two-step transformation process.
+Compiled is a build time [CSS-in-JS](https://reactjs.org/docs/faq-styling.html#what-is-css-in-js) library that has a two-step transformation process that outputs [atomic CSS](/atomic-css).
```
Comment (lorvsso) [inline packages/docs/src/pages/what-is-compiled.mdx:11]:
```suggestion
library with a two-step transformation process
that outputs [atomic CSS](/atomic-css).
```
avoid the double "that"

**Comment by lorvsso** [comment]
Site update is looking shmick!

File: packages/docs/src/components/side-nav.tsx (hunk 1 above)
Comment (madou) [reply packages/docs/src/components/side-nav.tsx:43]: fixed

**Pull request merged**